mllib/src/main/scala/org/apache/spark/ml/feature/FeatureHasher.scala

```diff
@@ -208,7 +208,8 @@ class FeatureHasher(@Since("2.3.0") override val uid: String) extends Transforme
       require(dataType.isInstanceOf[NumericType] ||
         dataType.isInstanceOf[StringType] ||
         dataType.isInstanceOf[BooleanType],
-        s"FeatureHasher requires columns to be of NumericType, BooleanType or StringType. " +
+        s"FeatureHasher requires columns to be of ${NumericType.simpleString}, " +
```

Comment:
In the original PR it doesn't seem like we rewrote the constant types only the dynamic ones (and this PR also doesn't seem to consistently rewrite the constant types referenced). What's the reason/how do you decide which ones you want to rewrite to ref simpleString?

Reply:
I think this PR rewrites always constant type referenced. I am not sure why you are saying it is not. If I missed some places, then it was just because I haven't seen them.

Reply:
I think that's my bad, looking through it I saw some raw type names but those are all in the suites which makes sense.